service/dap/server.go

```diff
@@ -1563,6 +1563,15 @@ func (s *Server) doStepCommand(command string, threadId int, asyncSetupDone chan
 		s.send(stopped)
 		return
 	}
+
+	if granularity == "instruction" {
```

Comment:
What does it mean to next vs step vs step out with instruction granularity?
What exactly does this look like in the UI?

Reply:
The specification of DAP doesn't say but I think that Next + instruction granularity is meant to step over a `CALL` instruction, like an assembly level debugger would do. But we don't have that feature.

Reply:
I think @aarzilli covered it, the details are not explicitly written out in the DAP spec, so I think we are free to make our own decisions, we can definitely look in to adding support for different behavior in each. I can add a TODO to investigate this at a future point.

The UI for all of these disassembly features is very new. Currently, if you are looking at disassembled code, then clicking the regular next, step in, step out buttons will include granularity="instruction".